### ISSUE TITLE
The implementation is different with NCF

### ISSUE THREAD

**Issue by familyld** [guoyang9/NCF#5]
Hi~
In your implementation, the embedding layers for both GMF and MLP use nn.Linear instead of nn.Embedding, these two modules are different, i.e., nn.Linear has an extra parameter (bias) which nn.Embedding has not. BTW, you also use RELU as the activation function to add non-linearity for these "embeddings". Such changes slightly increased the performance according to my experiments.

Another problem I occurred is about the negative sampling in NCF_input.py:

```
		if is_training:
			for k in neg_samples:
				feature_user.append(user)
				feature_item.append(k)
				labels_add.append(0)
		else:
			for k in neg_samples:
				feature_user.append(user)
				feature_item.append(k)
				labels_add.append(k)
```

I am confused about the last row `labels_add.append(k)`. Why k is appended instead of 0 for test data？

**Comment by guoyang9** [comment]
Thanks for commenting, labels are not used during test phrase. Therefore, it doesn't matter whether you append k or 0. I will use the Embedding api to replace the one-hot embedding matrix and implement other model settings on different datasets in a month (after I finish my recent work).

**Comment by familyld** [comment]
@guoyang9 Thanks. That would be great！

**Comment by guoyang9** [comment]
Please refer to the new implementation.
